web_app/hazard-reporting-map/components/modals/volunteers/AddVolunteer.tsx

```diff
@@ -0,0 +1,399 @@
+      }, 200);
+    }
+  }, [open, isClient]);
+
+  useEffect(() => {
+    if (!open && map) {
+      map.remove();
+      setMap(undefined);
+    }
+  }, [open, map]);
+  const handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { name, value } = e.target;
+    setFormInfo({ ...formInfo, [name]: value });
+  };
+
+  const handleDropDownChange = (value: string, name: string) => {
+    setFormInfo({ ...formInfo, [name]: value });
+  };
+
+  const handleFormSubmit = () => {
+    setIsFormLoading(true);
+    const promise = () =>
+      new Promise((resolve) =>
+        setTimeout(() => resolve(onOpenChange(!open)), 2000)
+      );
+
+    toast.promise(promise, {
+      loading: 'Wait a second...',
+      success: () => {
+        setIsFormLoading(false);
+        return `Volunteer added!`;
+      },
+      error: 'Error',
+    });
+  };
+
+  const handleDelete = () => {
+    setIsFormLoading(true);
+    const promise = () =>
+      new Promise((resolve) =>
+        setTimeout(() => resolve(onOpenChange(!open)), 2000)
+      );
+
+    toast.promise(promise, {
+      loading: 'Wait a second...',
+      success: () => {
+        setIsFormLoading(false);
+        return `Volunteer removed!`;
+      },
+      error: 'Error',
+    });
+  };
+  return (
+    <Drawer
+      open={open}
+      onOpenChange={onOpenChange}
+      onClose={() => onOpenChange(!open)}
+    >
+      <DrawerTrigger asChild>{children}</DrawerTrigger>
+      <DrawerContent>
+        <div className="h-[90vh] overflow-y-auto">
+          <Container className="mx-auto w-full ">
+            <DrawerHeader className=" w-full">
+              <DrawerTitle className="text-left font-semibold">
+                Add Volunteer
+              </DrawerTitle>
+              <div className="flex justify-end gap-3">
+                <Button
+                  onClick={handleDelete}
+                  variant={'danger'}
+                  className="px-8"
+                  disabled={isFormLoading}
+                >
+                  Delete
+                </Button>
+                <DrawerClose asChild>
+                  <Button
+                    variant="outline"
+                    className="bg-transparent rounded-none border-[#49454F] px-2 text-[#49454F]"
+                  >
+                    <X />
+                  </Button>
+                </DrawerClose>
+              </div>
+            </DrawerHeader>
+            <div className=" p-0 pb-0">
+              <div className="flex md:flex-row flex-col md:justify-between">
+                {/* inputs section */}
+                <div className="md:w-1/2 w-full  p-3 flex flex-col gap-4">
+                  {/* name, number, org */}
+                  <div className="w-full flex flex-col gap-5 ">
+                    <div className="flex flex-col lg:flex-row lg:items-center w-full  gap-2 2xl:gap-24 lg:gap-18">
+                      <Label className="w-[160px]  text-nowrap">
```

Comment:
use array mapping for similar and repeated components or elements .

Reply:
Made some reusable components instead of array mapping